ls1intum/Artemis
# Remove unnecessary restcall which caused problems

### Checklist
~- [x] Server: I added multiple integration tests (Spring) related to the features~
~- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)~
~- [x] Server: I implemented the changes with a good performance and prevented too many database calls~
~- [x] Client: I added multiple integration tests (Jest) related to the features~
~- [x] Client: I added `authorities` to all new routes and check the course groups for displaying~ navigation elements (links, buttons)
~- [x] Client: I added multiple screenshots/screencasts of my UI changes~
~- [x] Client: I translated all the newly inserted strings into German and English~

### Motivation and Context



### Description


### Steps for Testing

Focus on the student accounts for testing, but please check also the others(tutor, instructor, admin).

Testing on test server:
1. Log into Artemis
2. Navigate to Course Management
3. Select a course
4. Create a programming exercise
5. Navigate to course overview
6. Select the course 
7. Open the details page of the programming exercise

-> No error should occur here, check the network tab of the browser developer tools

Local testing:
1. Run artemis in your IDE
2. Login into Artemis
3. Navigate to Course Management
4. Select a course
5. Create a programming exercise without a connection to the vcs and ci 
-> normal creation procedure as for a normal programming exercise, but before you click on generate tick the checkbox "Setup without connection to the VCS and CI"

![Screenshot from 2020-04-30 18-23-16](https://user-images.githubusercontent.com/12066497/80735094-795b8500-8b10-11ea-8e3c-1fdcb8968453.png)

6. Navigate to course overview
7. Select the course
8. Open the details page of the programming exercise

-> No error should occur here, check the network tab of the browser developer tools
-> Two buttons (submission and result) should be visible
![Screenshot from 2020-04-30 18-27-55](https://user-images.githubusercontent.com/12066497/80735084-72347700-8b10-11ea-968d-0344957284b4.png)
